### DISCUSSION TITLE
Can I embed existing photo folders?

### DISCUSSION THREAD

**Discussion by rafo** [immich-app/immich#431] [Q&A] [answered]
Immich seems great for getting away from cloud-based providers like Google and Apple. However, I also have a lot of "normal" photos on my NAS, in neat folders.

```
Fotos
    2001
        010822_Maldives
            010822_1233_Airport.jpg
            ...
        010922_Seychelles
    2002
        020412_Polynesia
            020412_1011_Beach.dng
```

1. Can Immich integrate these (yet or in the future) in addition to the download folder?
2. Does Immich respect the file and folder structure and didnt alter them EDIT: Does not touch folder structure?

**Comment by alextran1502** [comment, marked as answer]
Hello, there is no support for this yet. I think we are moving closer to seriously considering this feature so stay tuned

**Reply by rafo** [reply]
> Hello, there is no support for this yet. I think we are moving closer to seriously considering this feature so stay tuned 

Good news!

What will happen if I point my immich download folder to my photos root? Does it traverse into subfolders?

**Reply by alextran1502** [reply]
> > Hello, there is no support for this yet. I think we are moving closer to seriously considering this feature so stay tuned 
> 
> Good news!
> 
> What will happen if I point my immich download folder to my photos root? Does it traverse into subfolders?

The Immich download folder is one direction only for now. It doesn't scan the existing files

**Comment by axgd-code** [comment]
> Immich seems great for getting away from cloud-based providers like Google and Apple. However, I also have a lot of "normal" photos on my NAS, in neat folders.
> 
> ```
> Fotos
>     2001
>         010822_Maldives
>             010822_1233_Airport.jpg
>             ...
>         010922_Seychelles
>     2002
>         020412_Polynesia
>             020412_1011_Beach.dng
> ```
> 
>     1. Can Immich integrate these (yet or in the future) in addition to the download folder?
> 
>     2. Does Immich respect the file and folder structure and didnt alter them EDIT: Does not touch folder structure?


I agree with you @rafo . The folder structure as album is a common way to organize library so it would be a high value feature. 
It's the first argument of Photoview : https://photoview.github.io/.

Both projects are so interesting 👍
